.github/workflows/bundleTest-thor.md

```diff
@@ -0,0 +1,353 @@
+```yaml
+on:
+  workflow_call:
+    inputs:
+      os:
+        type: string
+        description: 'Operating System'
+        required: false
+        default: 'ubuntu-22.04'
+      asset-name:
+        type: string
+        description: 'Asset Name'
+        required: false
+        default: 'build-docker-package'
+      dependencies:
+        type: string
+        description: 'Dependencies'
+        required: false
+        default: 'bison flex build-essential binutils-dev curl lsb-release libcppunit-dev python3-dev default-jdk r-base-dev r-cran-rcpp r-cran-rinside r-cran-inline pkg-config libtool autotools-dev automake git cmake xmlstarlet'
+      get-Stat: 
+        type: boolean
+        description: 'Run Query stat'
+        required: false
+        default: false
+      generate-Zap:
+        type: string
+        description: 'Generate ZAP files'
+        required: false
+        default: ''
+      test-core-file-generation: 
+        type: boolean
+        description: 'Test core file generation'
+        required: false
+        default: false
+```
+- **os** : Specifies the desired operating system of the runner machine using `runs-on: ${{ inputs.os }}`, with the default being Ubuntu-22.04.
+- **asset-name** : It specifies the name of the built artifact of the HPCC Platform. 
+- **dependencies** : Lists the required dependencies to install and start the platform.
+- **get-Stat** : This is boolean input that decides whether the QueryStat2.py step should be executed or not. The default value is false.
+- **generate-Zap** : Indicates if ZAP reports are needed for specific test cases. The default is an empty string.You can either pass the name of the test file with .ecl extension or * at the end. The later is used to refer the workuint name. Eg. "KMeansValidateOBT-240531-133100" can be refered as "KMeansValidateOBT*" 
+Example:   
+  Caller Workflow: [build-vcpkg.yml](https://github.com/hpcc-systems/HPCC-Platform/blob/master/.github/workflows/build-vcpkg.yml)
+    ```yaml
+    test-bundles-on-thor-ubuntu-22_04:
+        if: ${{ contains('pull_request,push', github.event_name) }}
+        needs: build-docker-ubuntu-22_04
+        uses: ./.github/workflows/bundleTest-thor.yml
+        with:
+          os: ubuntu-22.04
+          asset-name: 'docker-ubuntu-22_04'
+          generate-Zap: "'KMeansValidateOBT.ecl SVTest*'"
+        secrets: inherit
+    ```
+  **NOTE:** The double quotes are required if * notation is used to specify the name of the test. It makes sure that string function properly during the workflow run.
+- **test-core-file-generation** :A boolean input that determines whether the core file generation test step should be executed. It helps us to know whether the core files are generated as expected. 
```

Comment:
I think this help us to test whether our core handler working fine if a core file generated during tests.